docs/hoon/library/1a.md

```diff
@@ -64,7 +64,469 @@ Source
 
     ++  dec                                                 ::  decrement
       ~/  %dec
-      |=  a/@
+      |=  a=@
+      ~_  leaf+"decrement-underflow"
+      ?<  =(0 a)
+      =+  b=0
+      |-  ^-  @
+      ?:  =(a +(b))  b
+      $(b +(b))
+
+Examples
+--------
+
+    > (dec 7)
+    6
+    > (dec 0)
+    ! decrement-underflow
+    ! exit
+
+
+
+***
+### `++div`
+
+Divide
+
+Computes `a` divided by `b`.
+
+Accepts
+-------
+
+`a` is an atom.
+
+`b` is an atom.
+
+Produces
+--------
+
+An atom.
+
+Source
+------
+
+    ++  div                                                 ::  divide
+      ~/  %div
+      |:  [a=`@`1 b=`@`1]
+      :>  quotient
+      ^-  @
+      ~_  leaf+"divide-by-zero"
+      ?<  =(0 b)
+      =+  c=0
+      |-
+      ?:  (lth a b)  c
+      $(a (sub a b), c +(c))
+    ::
+
+Examples
+--------
+
+    > (div 4 2)
+    2
+    > (div 17 8)
+    2
+    > (div 20 30)
+    0
+
+
+***
+### `++gte`
+
+Greater-than/equal
+
+Tests whether `a` is greater than a number `b`.
+
+Accepts
+-------
+
+`a` is an atom.
+
+`b` is an atom.
+
+Produces
+--------
+
+A boolean.
+
+Source
+------
+
+    ++  gte                                                 ::  greater-equal
+      ~/  %gte
+      |=  [a=@ b=@]
+      ^-  ?
+      !(lth a b)
+
+
+Examples
+--------
+
+    > (gte 100 10)
+    %.y
+    > (gte 4 4)
+    %.y
+    > (gte 3 4)
+    %.n
+
+
+
+***
+### `++gth`
+
+Greater-than
+
+Tests whether `a` is greater than `b`.
+
+Accepts
+-------
+
+`a` is an atom.
+
+`b` is an atom.
+
+Produces
+--------
+
+A boolean.
+
+Source
+------
+
+    ++  gth                                                 ::  greater-than
+      ~/  %gth
+      |=  [a=@ b=@]
+      ^-  ?
+      !(lte a b)
+
+
+
+Examples
+--------
+
+    > (gth 'd' 'c')
+    %.y
+    > (gth ~h1 ~m61)
+    %.n
+
```

Comment:
These are good examples that show the flexibility of atoms, but there should be a third and fourth -- the first ones to read -- that just use raw atoms. Something like:

```
> (gth 42 41)
%.y

> (gth 42 43)
%.n

> (gth 'd' 'c')
%.y

> (gth ~h1 ~m61)
%.n
```